codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/AwsServiceModel.java

```diff
@@ -163,16 +169,90 @@ private MethodSpec getValueForField() {
     private List<MethodSpec> memberGetters() {
         return shapeModel.getNonStreamingMembers().stream()
                          .filter(m -> !m.getHttp().getIsStreaming())
-                         .map(m -> MethodSpec.methodBuilder(m.getFluentGetterMethodName())
-                                             .addJavadoc("$L", m.getGetterDocumentation())
-                                             .returns(typeProvider.fieldType(m))
-                                             .addModifiers(Modifier.PUBLIC)
-                                             .addCode(getterStatement(m))
-                                             .build())
+                         .flatMap(this::memberGetters)
                          .collect(Collectors.toList());
     }
 
-    private CodeBlock getterStatement(MemberModel model) {
+    private Stream<MethodSpec> memberGetters(MemberModel member) {
+        List<MethodSpec> result = new ArrayList<>();
+
+        if (shouldGenerateEnumGetter(member)) {
+            result.add(enumMemberGetter(member));
+        }
+
+        result.add(nonEnumMemberGetter(member));
```

Comment:
Name is kind of unintuitive as you'd call it for enums too it would just emit the string one.

Reply:
I'm trying to say "generate the non-enum getter for this member". For enums, that's a string. What name would you like me to use instead?

Reply:
I'd just call it memberGetter since it's the primary getter for any type of field, the enum one is the convenience one.

Reply:
:+1: